### PR TITLE
LIBITD-1083 Make Unit required for Unit level users

### PR DESCRIPTION
**Pull request by cfitz**
https://issues.umd.edu/browse/LIBITD-1083

### CODE REVIEW

**Comment by lib-ssdr-service** [comment]
Ok. Would you want to take a crack at this? I've got my hands full today
with this postgres stuff.

**Comment by dsteelma-umd** [comment]
@cfitz  - Not sure if you're done with the Postgres stuff yet, but if you are (and you want this issue back), then I did write up some additional integration tests that fail due to the scenarios outlined in my review comment.

I have pushed the tests up to the "feature/LIBITD-1083" branch in my fork: https://github.com/dsteelma-umd/annual-staffing-request/tree/feature/LIBITD-1083

**Comment by cfitz** [comment]
@dsteelma-umd Looks good. You should be able to push to this branch, so feel free to update the PR. 

I did notice a typo, which is I used && for an intersection, which in Ruby should be &. Also, for update, i think its checking the saved record in the policy rather than the incoming attributes, so I guess the controller should change the attributes first then run the authorize the record.

File: app/policies/request_policy.rb [hunk 1]
```diff
@@ -1,10 +1,16 @@
+class NoUnitForUnitUser < StandardError; end
+
 class RequestPolicy < ApplicationPolicy
   def create?
-    true
+    return true if @user.admin? || @record.is_a?(Class)
+    if @record.unit_id.nil? && @user.org_types.none? { |o| o != 'unit' }
```
Comment (cfitz) [inline app/policies/request_policy.rb]: Should be "&" not "&&"

File: app/policies/request_policy.rb [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+      raise NoUnitForUnitUser, 'You must specific a unit'
+    end
+    (@user.active_organizations.map(&:id) && [@record.organization_id, @record.unit_id]).any?
```
Comment (cfitz) [inline app/policies/request_policy.rb]: Same here. Should be "&" not "&&"

**Comment by dsteelma-umd** [comment]
@cfitz - I've cherry-picked the integration tests into the feature/LIBITD-1083 branch. Note that these tests currently FAIL. I've reassigned the Jira issue back to you.

**Comment by cfitz** [comment]
@dsteelma-umd heh, ok... I made the changes and rebased the PR.

**Pull request merged**